src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,26 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.impl.response.UserResponseMapper;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserService userService;
+    private final UserResponseMapper userResponseMapper;
+
+    public UserController(UserService userService, UserResponseMapper userResponseMapper) {
+        this.userService = userService;
+        this.userResponseMapper = userResponseMapper;
+    }
+
+    @GetMapping("by-email/{/email}")
+    public UserResponseDto getByEmail(@PathVariable String email) {
+        return userResponseMapper.toDto(userService.findByEmail(email).get());

```

Comment:
Directly using .get() on Optional can lead to NoSuchElementException if UserService#findByEmail doesn't find a user. You should handle the case when a user with a provided email is not found.